src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -1081,11 +1081,11 @@ public void getModifiedDocIds(DocIdPusher pusher)
                 log.log(Level.FINE, "latestTimestamp updated: {0}",
                     formatter.format(new Date(latestTimestamp.getTime())));
               }
-              if (lastDocIdTimestamp == null || ts.after(lastDocIdTimestamp)) {
-                lastDocIdTimestamp = ts;
-                docIdLastModifiedMap.put(id, lastDocIdTimestamp);
-                log.log(Level.FINE, "lastDocIdTimestamp updated: {0}",
-                    formatter.format(new Date(lastDocIdTimestamp.getTime())));
+              if (cachedLastModified == null) {
```

Comment:
This if statement is wrong. Why would a row have to be uncached in order to update the cache with a new last modified?